CONTRIBUTING.md

```diff
@@ -163,3 +163,154 @@ minor versions at this time.
+compiler. To work on Sensu, you will need the latest release of Go.
+
+[Go installation instructions](https://golang.org/doc/install)
+
+### Protobuf
+
+#### Overview
+
+We are using the version **proto3** of the protocol buffers language. Here are some useful resources:
+
+[To learn more about protocol buffers](https://developers.google.com/protocol-buffers/docs/overview)
+
+[The proto3 language guide](https://developers.google.com/protocol-buffers/docs/proto3)
+
+#### Installation
+
+Install the protobuf compiler since we don't use the one that golang uses.
+```
+brew install protobuf
+```
+Otherwise, see the **for non-C++ users** [instructions here.](https://github.com/google/protobuf#protocol-compiler-installation)
+
+#### Quick Start
+
+Once you make a change to any `*.proto` file within the **types** package, you will need to regenerate the associated `*.pb.go` file. To do so, simply run `go generate` on the package.
+
+### Dependencies
+
+Sensu uses [golang/dep](https://github.com/golang/dep) for managing its
+dependencies. You will need to install the latest stable version of dep in
+order to install Sensu's dependencies.
+
+[Dep releases](https://github.com/golang/dep/releases)
+
+#### Usage
+
+Running the following will pull all required dependencies, including static
+analysis and linter tools.
+
+```shell
+./build.sh deps
+```
+
+Later, if you would like to add a dependency, run:
+
+```shell
+dep ensure -add https://my-repo.com/my/dep
+```
+
+If you would like to update a dependency, run:
+
+```shell
+dep ensure -update https://my-repo.com/my/dep
+```
+
+When you would like to remove a dependency, remove the it from `Gopkg.toml` and
+then run:
+
+```shell
+dep prune
+```
+
+#### Further Reading
+
+- [The Saga of Go Dependency Management](https://blog.gopheracademy.com/advent-2016/saga-go-dependency-management/)
+- [`dep` Usage](https://github.com/golang/dep#usage)
+
+## Building
+
+### Docker
+
+The simplest way to the build Sensu is with the `sensu-go-build` image. The
+image contains all the required tools to build the agent, backend and sensuctl.
+
+```sh
+docker pull sensu/sensu-go-build
+docker run -it -e GOOS=darwin -v `pwd`:/go/src/github.com/sensu/sensu-go --entrypoint='/go/src/github.com/sensu/sensu-go/build.sh' sensu/sensu-go-build
+```
+
+If you would like to build for different platforms and architectures use GOOS
+and GOARCH env variables. See [Optional environment variables](https://golang.org/doc/install/source#environment) for more.
+
+When complete your binaries will be present in the `target` directory.
+
+### Manually
+
+First ensure that you have the required tools installed to build the programs.
+
+* Ensure that you have the Go tools installed and your environment configured.
+  If not follow the official
+  [Install the Go tools](https://golang.org/doc/install#install) guide.
+* When building the Sensu backend you will need NodeJS and Yarn installed so
```

Comment:
You don't need it really, but you'll get a warning and won't be able to use the dashboard. It will also build the backend faster!

Reply:
Let's not mention that :)